picard/track.py

```diff
@@ -64,6 +66,52 @@
 }
 
 
+class TagGenreFilter:
+
+    def __init__(self, setting=None):
+        if setting is None:
+            setting = config.setting
+        self.genres_filter = setting['genres_filter']
+
+        self.errors = dict()
+        self.match_regexes = defaultdict(list)
+        for lineno, line in enumerate(self.genres_filter.splitlines()):
+            line = line.strip()
+            if line and line[0] in ('+', '-'):
+                _list = line[0]
+                remain = line[1:].strip()
+                if not remain:
+                    continue
+                if len(remain) > 2 and remain[0] == '/' and remain[-1] == '/':
+                    remain = remain[1:-1]
+                    try:
+                        regex_search = re.compile(remain, re.IGNORECASE)
+                    except Exception as e:
+                        log.error("Failed to compile regex /%s/: %s", remain, e)
+                        self.errors[lineno] = str(e)
+                        regex_search = None
+                else:
+                    # FIXME?: only support '*' (not '?' or '[abc]')
+                    # replace multiple '*' by one
+                    star = re.escape('*')
+                    remain = re.sub(star + '+', '*', remain)
+                    regex = '.*'.join([re.escape(x) for x in remain.split('*')])
+                    regex_search = re.compile('^' + regex + '$', re.IGNORECASE)
+                if regex_search:
+                    self.match_regexes[_list].append(regex_search)
+
+    def skip(self, tag):
```

Comment:
Why not provide a `filter` method which accepts a list and returns only valid genres? Again this is with using this class in plugins in mind, it would make a nicer interface.

Reply:
Done in 1b08c55de545b7